### `Importing important libraries`

In [10]:
import pandas as pd
from matplotlib import pyplot as plt

# setting pandas option
pd.set_option("display.max_columns", 66)

### `Importing Raw Data and displaying few rows`

##### *Fault Data*

In [ ]:
fault_df = pd.read_csv("../../data/raw/fault_data.csv")

fault_df.head(5) # Displaying the first 5 rows

fault_df['Fault'].value_counts()

Fault
FF    254
EF    174
AF     62
GF     43
MF     20
Name: count, dtype: int64

- gf: generator heating fault
- mf: mains failure fault
- ff: feeding fault
- af: air cooling fault
- ef: excitation fault

##### *Scada Data*

In [16]:
scada_df = pd.read_csv('../../data/raw/scada_data.csv')

scada_df.head(5) # Displaying the first 5 rows

DateTime        Time  Error  WEC: ava. windspeed  WEC: max. windspeed  \
0  5/1/2014 0:00  1398920448      0                  6.9                  9.4   
1  5/1/2014 0:09  1398920960      0                  5.3                  8.9   
2  5/1/2014 0:20  1398921600      0                  5.0                  9.5   
3  5/1/2014 0:30  1398922240      0                  4.4                  8.3   
4  5/1/2014 0:39  1398922752      0                  5.7                  9.7   

   WEC: min. windspeed  WEC: ava. Rotation  WEC: max. Rotation  \
0                  2.9                 0.0                0.02   
1                  1.6                 0.0                0.01   
2                  1.4                 0.0                0.04   
3                  1.3                 0.0                0.08   
4                  1.2                 0.0                0.05   

   WEC: min. Rotation  WEC: ava. Power  WEC: max. Power  WEC: min. Power  \
0                 0.0                0                0                0   
1                 0.0                0                0                0   
2                 0.0                0                0                0   
3                 0.0                0                0                0   
4                 0.0                0                0                0   

   WEC: ava. Nacel position including cable twisting  WEC: Operating Hours  \
0                                                640                   881   
1                                                640                   881   
2                                                640                   881   
3                                                641                   881   
4                                                633                   881   

   WEC: Production kWh  WEC: Production minutes  WEC: ava. reactive Power  \
0               939053                        6                         0   
1               939053                        6                         0   
2               939053                        6                         0   
3               939053                        6                         0   
4               939053                        6                         0   

   WEC: max. reactive Power  WEC: min. reactive Power  \
0                         0                         0   
1                         0                         0   
2                         0                         0   
3                         0                         0   
4                         0                         0   

   WEC: ava. available P from wind  WEC: ava. available P technical reasons  \
0                              838                                        0   
1                              443                                        0   
2                              525                                        0   
3                              250                                        0   
4                              549                                        0   

   WEC: ava. Available P force majeure reasons  \
0                                          838   
1                                          443   
2                                          525   
3                                          250   
4                                          549   

   WEC: ava. Available P force external reasons  WEC: ava. blade angle A  \
0                                           838                91.959999   
1                                           443                91.959999   
2                                           525                91.940002   
3                                           250                91.949997   
4                                           549                91.940002   

   Sys 1 inverter 1 cabinet temp.  Sys 1 inverter 2 cabinet temp.  \
0                              24                              26   
1                    

##### *Status Data*

In [17]:
status_df = pd.read_csv('../../data/raw/status_data.csv')

status_df.head(5) # Displaying the first 5 rows

Time  Main Status  Sub Status Full Status  \
0  24/04/2014 12:37:38            0           0       0 : 0   
1  25/04/2014 19:27:24           71         104    71 : 104   
2  26/04/2014 09:30:13            8           0       8 : 0   
3  26/04/2014 10:05:19            8           0       8 : 0   
4  26/04/2014 10:05:25            8           0       8 : 0   

                                         Status Text  T  Service  FaultMsg  \
0                               Turbine in operation  1    False     False   
1  Insulation monitoring : Insulation fault Phase U2  6    False      True   
2                                        Maintenance  6     True     False   
3                                        Maintenance  6    False     False   
4                                        Maintenance  6     True     False   

   Value0  
0     7.4  
1    20.5  
2    17.1  
3     8.7  
4    10.6

### `Basic summary of the imported data`

#### *Fault Dataset*

##### *Shape of the dataset*

In [18]:
fault_df.shape

(553, 3)

The fault dataset has:

- 3 columns
- 533 rows of information

##### *Column types*

In [22]:
fault_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 553 entries, 0 to 552
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   DateTime  553 non-null    object 
 1   Time      553 non-null    float64
 2   Fault     553 non-null    object 
dtypes: float64(1), object(2)
memory usage: 13.1+ KB


The fault dataset has a column called "DataTime" which is of object type. This column will have to be converted into datatime dtype.

##### *Any Duplicates?*

In [24]:
fault_df.duplicated().sum()

0

No duplicated data in the fault dataset

##### *Any Missing Data?*

In [25]:
fault_df.isna().sum()

DateTime    0
Time        0
Fault       0
dtype: int64

No missing data in our fault dataset

#### *Scada Dataset*

##### *Shape of the dataset*

In [19]:
scada_df.shape

(49027, 66)

The scada dataset has:

- 66 columns
- 49,027 rows of information

##### *Column types*

In [21]:
scada_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49027 entries, 0 to 49026
Data columns (total 66 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   DateTime                                           49027 non-null  object 
 1   Time                                               49027 non-null  int64  
 2   Error                                              49027 non-null  int64  
 3   WEC: ava. windspeed                                49027 non-null  float64
 4   WEC: max. windspeed                                49027 non-null  float64
 5   WEC: min. windspeed                                49027 non-null  float64
 6   WEC: ava. Rotation                                 49027 non-null  float64
 7   WEC: max. Rotation                                 49027 non-null  float64
 8   WEC: min. Rotation                                 49027 non-null  float64
 9   WEC: a

Similarly like the fault dataset, the "DateTime" column in scada dataset will need to be converted into datetime format. Also, this dataset has a "Time" column will also be needed to convert into a time format.

##### *Any Duplicates?*

In [30]:
scada_df.duplicated().sum()

0

There are no duplicates in the scada dataset

##### *Any Missing Data?*

In [33]:
scada_df.isna().sum()

DateTime                 0
Time                     0
Error                    0
WEC: ava. windspeed      0
WEC: max. windspeed      0
                        ..
Control cabinet temp.    0
Transformer temp.        0
RTU: ava. Setpoint 1     0
Inverter averages        0
Inverter std dev         0
Length: 66, dtype: int64

There is no missing data in the scada dataset

#### *Status Dataset*

##### *Shape of the dataset*

In [20]:
status_df.shape

(1849, 9)

The status dataset has:

- 9 columns
- 1,849 rows of information

##### *Column types*

In [23]:
status_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1849 entries, 0 to 1848
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Time         1849 non-null   object 
 1   Main Status  1849 non-null   int64  
 2   Sub Status   1849 non-null   int64  
 3   Full Status  1849 non-null   object 
 4   Status Text  1849 non-null   object 
 5   T            1849 non-null   int64  
 6   Service      1849 non-null   bool   
 7   FaultMsg     1849 non-null   bool   
 8   Value0       1849 non-null   float64
dtypes: bool(2), float64(1), int64(3), object(3)
memory usage: 104.9+ KB


The "Time" column will need to be converted into an appropriate dtype.

##### *Any Duplicates?*

In [36]:
status_df.duplicated().sum()

0

There is no duplicated data in our status dataset

##### *Any Missing Data?*

In [35]:
status_df.isna().sum()

Time           0
Main Status    0
Sub Status     0
Full Status    0
Status Text    0
T              0
Service        0
FaultMsg       0
Value0         0
dtype: int64

There is no missing data in our status dataset